File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Java/service/DeliveryService.java
package service;

import models.DeliveryPerson;
import utils.DeliveryStatus;

import java.util.HashMap;
import java.util.Map;

public class DeliveryService {
    private Map<String, DeliveryPerson> deliveryPersons = new HashMap<>();

    public void registerDeliveryPerson(DeliveryPerson person) {
        deliveryPersons.put(person.getId(), person);
    }

    public DeliveryPerson getAvailableDeliveryPerson() {
        for (DeliveryPerson person : deliveryPersons.values()) {
            if (person.getStatus() == DeliveryStatus.AVAILABLE) {
                return person;
            }
        }
        return null;
    }
}

File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Cpp/service/src/DeliveryService.cpp
#include "../include/DeliveryService.h"

void DeliveryService::registerDeliveryPerson(const std::shared_ptr<DeliveryPerson>& person) {
    deliveryPersons[person->getId()] = person;
}

std::shared_ptr<DeliveryPerson> DeliveryService::getAvailableDeliveryPerson() const {
    for (const auto& pair : deliveryPersons) {
        const auto& person = pair.second;
        if (person->getStatus() == DeliveryStatus::AVAILABLE) {
            return person;
        }
    }
    return nullptr;
}